activerecord/test/cases/validations/uniqueness_validation_test.rb

```diff
@@ -386,4 +386,21 @@ def test_validate_uniqueness_on_empty_relation
     topic = TopicWithUniqEvent.new
     assert topic.valid?
   end
+
+  def test_does_not_validate_uniqueness_of_if_parent_record_is_validate_false
+    Reply.validates_uniqueness_of(:content)
+
+    Reply.create!(content: "Topic Title")
+
+    reply = Reply.new(content: "Topic Title")
+    reply.save!(validate: false)
+    assert reply.persisted?
+
+    topic = Topic.new(reply_ids: [reply.id])
+    topic.save!
+
+    assert_equal topic.replies.size, 1
+    assert reply.valid?
```

Comment:
Understood thanks for the response. It's a big change for me. Then some
validations will run and others don't when the record has no changes.
Anyway I always add a condition on the uniqueness validations to avoid
extra queries
On Feb 7, 2015 5:59 PM, "Jason Nochlin" <notifications@github.com> wrote:

> In activerecord/test/cases/validations/uniqueness_validation_test.rb
> <https://github.com/rails/rails/pull/18612#discussion_r24290679>:
>
> > @@ -386,4 +386,21 @@ def test_validate_uniqueness_on_empty_relation
> >      topic = TopicWithUniqEvent.new
> >      assert topic.valid?
> >    end
> > +
> > +  def test_does_not_validate_uniqueness_of_if_parent_record_is_validate_false
> > +    Reply.validates_uniqueness_of(:content)
> > +
> > +    Reply.create!(content: "Topic Title")
> > +
> > +    reply = Reply.new(content: "Topic Title")
> > +    reply.save!(validate: false)
> > +    assert reply.persisted?
> > +
> > +    topic = Topic.new(reply_ids: [reply.id])
> > +    topic.save!
> > +
> > +    assert_equal topic.replies.size, 1
> > +    assert reply.valid?
>
> As an example, if you change a different attribute, say title, then the
> validations will be run:
>
>   def test_does_not_validate_uniqueness_of_if_parent_record_is_validate_false
>     ...
>     assert_equal topic.replies.size, 1
>     assert reply.valid?
>     reply.title = "Change An Attribute"
>     assert reply.valid? #=> Fails!
>     ...
>   end
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/18612/files#r24290679>.
>